src/libostree/ostree-repo.c

```diff
@@ -3113,6 +3113,53 @@ reload_remote_config (OstreeRepo          *self,
   return TRUE;
 }
 
+static gboolean
+reload_sysroot_config (OstreeRepo          *self,
+                       GCancellable        *cancellable,
+                       GError             **error)
+{
+  { g_autofree char *bootloader = NULL;
+
+    if (!ot_keyfile_get_value_with_default (self->config, "sysroot",
+                                            "bootloader", "auto",
+                                            &bootloader, error))
+      {
+        /* XXX: quick hack to make existing config files not fail
+         * on not having the [sysroot] group present. */
+        if (error)
```

Comment:
The pattern for this normally is creating a local_error so you can always check for the exact error, and then `g_propagate_error` if needed.

Reply:
Got it - makes sense.